lib/clientv2/node.go

```diff
@@ -92,10 +92,10 @@ func (r nodes) Delete(ctx context.Context, name string, opts options.DeleteOptio
 	}
 	_, err = r.client.IPAM().ReleaseIPs(context.Background(), ips)
 	if err != nil {
-		if _, ok := err.(errors.ErrorResourceDoesNotExist); !ok {
-			if _, ok := err.(errors.ErrorOperationNotSupported); !ok {
-				return nil, err
-			}
+		switch err.(type) {
+		case nil, errors.ErrorResourceDoesNotExist, errors.ErrorOperationNotSupported:
```

Comment:
Since you're checking for `nil` here, this doesn't need to be inside an `if err != nil` clause